src/Pos/TxHistory.hs

```diff
@@ -0,0 +1,241 @@
+{-# LANGUAGE CPP                  #-}
+{-# LANGUAGE InstanceSigs         #-}
+{-# LANGUAGE RankNTypes           #-}
+{-# LANGUAGE ScopedTypeVariables  #-}
+{-# LANGUAGE TemplateHaskell      #-}
+{-# LANGUAGE TypeFamilies         #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Pos.TxHistory
```

Comment:
`Pos.Txp.TxHistory` is better name, I think. Again, better to put inside some subpart, and `Txp` seems like reasonable part here.